app/Jobs/MarkThreadSolution.php

```diff
@@ -25,6 +26,6 @@ public function __construct(Thread $thread, Reply $solution)
 
     public function handle()
     {
-        $this->thread->markSolution($this->solution);
+        $this->thread->markSolution($this->solution, Auth::user());
```

Comment:
Please provide the user through the constructor of the job. In general you shouldn't access global state in a command/job.